### PR TITLE
Introduce `wp_cache_add_redis_hash_groups()`

### LINKED ISSUE #113: Allow only some group to delete whole group deletes
When WP_REDIS_USE_CACHE_GROUPS is set to true, then a hash set is used to store keys. However this means that the expires field is then ignored, a ttls are ignored. However there are valid reasons to have ttl on caches and not invalidating them can break many things. There are examples in core where caches should invalidate themselves. Good example is [here](https://github.com/WordPress/WordPress/blob/1036637afcae3725c64a65f8e7a334bb4a94da7d/wp-includes/class-wp-term-query.php#L735). 

A new method be added (similar to wp_cache_add_non_persistent_groups), that allows a developers to define which groups should be allowed to be delete and which group still respect the expire time. 

For example, I would want the cache group of "posts" to allow to deleted, by the terms or batcache group not.

### PR DESCRIPTION
**Pull request by danielbachhuber**
This function permits registering specific groups to use Redis hashes,
and is more precise than our existing `WP_REDIS_USE_CACHE_GROUPS` constant.

Fixes #113

### CODE REVIEW

**Comment by danielbachhuber** [comment]
@stevector #reviewmerge

File: object-cache.php [hunk 1]
```diff
@@ -403,6 +422,18 @@ public function add_non_persistent_groups( $groups ) {
 	}
 
 	/**
+	 * Sets the list of groups that use Redis hashes.
+	 *
+	 * @param array $groups List of groups that use Redis hashes.
+	 */
+	public function add_redis_hash_groups( $groups ) {
+		$groups = (array) $groups;
```
Comment (stevector) [inline object-cache.php]: why cast to an array when we could type hint?

File: object-cache.php [hunk 2]
```diff
@@ -937,6 +968,19 @@ protected function _should_persist( $group ) {
 	}
 
 	/**
+	 * Should this group use Redis hashes?
+	 *
+	 * @param string $group Cache group.
+	 * @return bool True if the group should use Redis hashes, false if not.
+	 */
+	protected function _should_use_redis_hashes( $group ) {
```
Comment (stevector) [inline object-cache.php]: @danielbachhuber you recently had a PR that removed the preceding `_` for a protected function. Let's stick with that convention.

File: object-cache.php (hunk 1 above)
Comment (danielbachhuber) [inline object-cache.php]: > why cast to an array when we could type hint?

Copied code from the other method.

File: object-cache.php (hunk 2 above)
Comment (danielbachhuber) [inline object-cache.php]: > Let's stick with that convention.

Ok — let's do in a follow-up PR.

**Comment by danielbachhuber** [comment]
@stevector Can we go ahead and land this, so we can release a v0.6.0?

**Comment by stevector** [comment]
Yeah, I had pinged @spacedmonkey over in https://github.com/pantheon-systems/wp-redis/issues/113 hoping to get more testing. But I think we can just merge.

**Pull request merged**